ts/features/messages/hooks/useAttachmentDownload.tsx

```diff
@@ -37,37 +75,28 @@ export const useAttachmentDownload = (
       void mixpanelTrack("PN_ATTACHMENT_DOWNLOADFAILURE");
       showToast(i18n.t("messageDetails.attachments.failing.details"));
     } else if (download) {
-      const path = download.path;
-      const attachment = download.attachment;
+      const { path, attachment } = download;
+
       if (attachment.contentType === ContentTypeValues.applicationPdf) {
         openPreview(attachment);
       } else {
-        if (isIos) {
-          ReactNativeBlobUtil.ios.presentOptionsMenu(path);
-        } else {
-          try {
-            const downloadFilePath =
-              await ReactNativeBlobUtil.MediaCollection.copyToMediaStore(
-                {
-                  name: attachment.displayName,
-                  parentFolder: "",
-                  mimeType: attachment.contentType
-                },
-                "Download",
-                path
-              );
-
-            await ReactNativeBlobUtil.android.addCompleteDownload({
-              mime: attachment.contentType,
-              title: attachment.displayName,
-              showNotification: true,
-              description: attachment.displayName,
-              path: downloadFilePath
-            });
-          } catch (e) {
-            showToast(i18n.t("messageDetails.attachments.failing.details"));
-          }
-        }
+        await pipe(
+          isAndroid,
+          O.fromPredicate(identity),
+          TE.fromOption(() => undefined),
+          TE.mapLeft(() => ReactNativeBlobUtil.ios.presentOptionsMenu(path)),
+          TE.chain(_ =>
+            pipe(
+              taskCopyToMediaStore(attachment, path),
+              TE.chain(downloadFilePath =>
+                taskAddCompleteDownload(attachment, downloadFilePath)
+              ),
+              TE.mapLeft(_ =>
+                showToast(i18n.t("messageDetails.attachments.failing.details"))
+              )
+            )
+          )
+        )();
```

Comment:
Is the async/await required? What happen removing `async/await` and using `then/catch` in the pipe instead? What do you think about?

```suggestion
        pipe(
          isAndroid,
          O.fromPredicate(identity),
          TE.fromOption(() => undefined),
          TE.mapLeft(() => ReactNativeBlobUtil.ios.presentOptionsMenu(path)),
          TE.chain(_ =>
            pipe(
              taskCopyToMediaStore(attachment, path),
              TE.chain(downloadFilePath =>
                taskAddCompleteDownload(attachment, downloadFilePath)
              ),
              TE.mapLeft(_ =>
                showToast(i18n.t("messageDetails.attachments.failing.details"))
              )
            )
          )
        )().catch(//...);
```